src/types.ts

```diff
@@ -15,14 +16,51 @@ export const chainExplorerSchema = z.object({
   browserUrl: z.string().url(),
 });
 
+export const chainProviderSchema = z
+  .object({
+    alias: z.string(),
+    homepageUrl: z.string().url().optional(),
+    rpcUrl: z.string().url().optional(),
+  })
+  .refine(
+    // Either rpcUrl or homepageUrl must be present
+    (provider) => provider.rpcUrl || provider.homepageUrl,
+    { message: 'rpcUrl or homepageUrl is required' }
+  );
+
+export const chainProvidersSchema = z.array(chainProviderSchema).superRefine((providers, ctx) => {
```

Comment:
Some more validations here:

1. Each chain must have at least 1 provider with `alias === 'default'`
2. `alias` values must be unique
3. providers with `alias === 'default'` or `alias === 'public` must also have an `rpcUrl` value

Reply:
[Agreed](https://github.com/api3dao/chains/pull/85#discussion_r1376078930)